internal/engine/ingester/git/git.go

```diff
@@ -105,47 +99,72 @@ func (gi *Git) Ingest(ctx context.Context, ent protoreflect.ProtoMessage, params
 
 	return &interfaces.Result{
 		Object:     nil,
-		Fs:         wt.Filesystem,
-		Storer:     r.Storer,
+		Fs:         fs,
+		Storer:     storer,
 		Checkpoint: chkpoint,
 	}, nil
 }
 
-func (gi *Git) getBranch(ent protoreflect.ProtoMessage, userCfg *IngesterConfig) string {
-	// If the user has specified a branch, use that
-	if userCfg.Branch != "" {
-		return userCfg.Branch
+func (gi *Git) ingestPullRequest(
+	ctx context.Context, ent *pbinternal.PullRequest, params map[string]any) (*interfaces.Result, error) {
+	userCfg := &IngesterConfig{}
+	if err := mapstructure.Decode(params, userCfg); err != nil {
+		return nil, fmt.Errorf("failed to read git ingester configuration from params: %w", err)
 	}
 
-	// If the branch is provided in the rule-type
-	// configuration, use that
-	if gi.cfg.Branch != "" {
-		return gi.cfg.Branch
+	if ent.GetBaseCloneUrl() == "" || ent.GetBaseRef() == "" {
+		return nil, fmt.Errorf("could not get PR base branch %q from %q", ent.GetBaseRef(), ent.GetBaseCloneUrl())
+	}
+	if ent.GetTargetCloneUrl() == "" || ent.GetTargetRef() == "" {
+		return nil, fmt.Errorf("could not get PR target branch %q from %q", ent.GetTargetRef(), ent.GetTargetCloneUrl())
 	}
 
-	if repo, ok := ent.(*pb.Repository); ok {
-		if repo.GetDefaultBranch() != "" {
-			return repo.GetDefaultBranch()
-		}
-	} else if pr, ok := ent.(*pbinternal.PullRequest); ok {
-		return pr.GetTargetRef()
+	baseFs, _, _, err := gi.fetchClone(ctx, ent.GetBaseCloneUrl(), ent.GetBaseRef())
+	if err != nil {
+		return nil, fmt.Errorf("failed to clone base branch %s from %s: %w", ent.GetBaseRef(), ent.GetBaseCloneUrl(), err)
+	}
+	targetFs, storer, head, err := gi.fetchClone(ctx, ent.GetTargetCloneUrl(), ent.GetTargetRef())
+	if err != nil {
+		return nil, fmt.Errorf("failed to clone target branch %s from %s: %w", ent.GetTargetRef(), ent.GetTargetCloneUrl(), err)
 	}
 
-	// If the branch is not provided in the rule-type
-	// configuration, use the default branch
-	return defaultBranch
+	checkpoint := checkpoints.NewCheckpointV1Now().WithBranch(ent.GetTargetRef()).WithCommitHash(head.Hash().String())
```

Comment:
We don't need to fix this in this PR... But... I wonder if we should have a base and target checkpoints checksum checkpoints?

Reply:
Since "base" is the branch that the PR will be merged into, I think that only changes if the target/head branch changes.